app/assets/stylesheets/map_marker.scss

```diff
@@ -0,0 +1,98 @@
+$icon-height:    20px;
+$icon-height-hovered: 30px;
+
+$map-marker-left: 10px;
+$map-marker-left-hover: 15px;
+
+$map-marker-text: 16px;
+$map-marker-text-hover: 24px;
+
+$hover-animation-duration: 100ms;
+
+@mixin hover-animation() {
+  -webkit-transition: $hover-animation-duration; /* Safari */
+  transition: $hover-animation-duration;
+  transition-timing-function: ease-in;
```

Comment:
I think you're missing the properties to transition.
See: https://developer.mozilla.org/en-US/docs/Web/CSS/transition